src/lib.rs

```diff
@@ -39,4 +62,76 @@ mod tests {
         // THEN it should be zero
         assert_eq!(size, 0);
     }
-}
+
+    #[test]
+    fn it_should_report_the_correct_number_of_elements_when_filled() {
```

Comment:
I like your naming convention and I really love that you followed the GIVEN-WHEN-THEN approach :heart_eyes: